pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -225,22 +231,40 @@ func (r *ControlPlaneReconciler) Reconcile() (result reconcile.Result, err error
 		return
 	}
 
-	if r.Status.ObservedGeneration == 0 {
-		reconciliationMessage = fmt.Sprintf("Successfully installed generation %d", r.Instance.GetGeneration())
-		r.Manager.GetRecorder(controllerName).Event(r.Instance, "Normal", "ServiceMeshInstalled", reconciliationMessage)
-	} else {
+	if r.isUpdating() {
 		reconciliationMessage = fmt.Sprintf("Successfully updated from generation %d to generation %d", r.Status.ObservedGeneration, r.Instance.GetGeneration())
-		r.Manager.GetRecorder(controllerName).Event(r.Instance, "Normal", "ServiceMeshUpdated", reconciliationMessage)
+		r.Manager.GetRecorder(controllerName).Event(r.Instance, corev1.EventTypeNormal, eventReasonUpdated, reconciliationMessage)
+	} else {
+		reconciliationMessage = fmt.Sprintf("Successfully installed generation %d", r.Instance.GetGeneration())
+		r.Manager.GetRecorder(controllerName).Event(r.Instance, corev1.EventTypeNormal, eventReasonInstalled, reconciliationMessage)
 	}
 	r.Status.ObservedGeneration = r.Instance.GetGeneration()
 	updateReconcileStatus(&r.Status.StatusType, nil)
 
+	_, err = r.updateReadinessStatus()
+
 	r.Log.Info("Completed ServiceMeshControlPlane reconcilation")
-	// requeue to ensure readiness is updated
-	result.Requeue = true
 	return
 }
 
+func (r *ControlPlaneReconciler) pauseReconciliation(chartName string, err error) (string, error) {
+	var reason string
+	if r.isUpdating() {
```

Comment:
might be cleaner to say: 

reason := eventReasonInstall
if r.IsUpdating() {
  reason = eventReasonPausingUpdate
}

Reply:
Saves resetting the variable.  I guess it depends on preference.